tests/src/test/java/alluxio/client/fs/UfsSyncIntegrationTest.java

```diff
@@ -605,6 +606,73 @@ public void recursiveSync() throws Exception {
     assertFalse(mFileSystem.exists(new AlluxioURI(alluxioPath(fileA))));
   }
 
+  @LocalAlluxioClusterResource.Config(
+      confParams = {
+          PropertyKey.Name.USER_FILE_METADATA_LOAD_TYPE, "NEVER"
+      })
+  @Test
+  public void recursiveSyncCacheDescendants() throws Exception {
+    // make nested directories/files in UFS
+    new File(ufsPath("/dir1")).mkdirs();
+    new File(ufsPath("/dir1/dir2")).mkdirs();
+    new File(ufsPath("/dir1/dir2/dir3")).mkdirs();
+    String fileA = "/dir1/dir2/dir3/fileA";
+    String fileB = "/dir1/dir2/dir3/fileB";
+    String fileNew = "/dir1/dir2/dir3/fileNew";
+    writeUfsFile(ufsPath(fileA), 1);
+    writeUfsFile(ufsPath(fileB), 1);
+
+    FileSystemMasterCommonPOptions longinterval =
+        FileSystemMasterCommonPOptions.newBuilder().setSyncIntervalMs(Constants.HOUR_MS).build();
+
+    // Should not exist, since no loading or syncing
+    assertFalse(mFileSystem.exists(new AlluxioURI(alluxioPath(fileA)), ExistsPOptions.newBuilder()
+        .setCommonOptions(PSYNC_NEVER).build()));
+
+    try {
```

Comment:
is the standard practice to use the `ExpectedException` rule here?

Reply:
It was, but then it was relaxed at a later time. I don't really like that rule, since you can only use 1 expected exception per method.

Reply:
sounds good